src/components/App.js

```diff
@@ -108,45 +100,49 @@ class App extends Component {
       pincodeRequired,
       previousUsername } = this.props;
 
-    const displaySettings = !pincodeRequired && (showUsernameModal || username === '');
+    let { showUsernameModal } = this.state;
+    showUsernameModal = !pincodeRequired && (showUsernameModal || username === '');
+
     const chatInputFocus = !pincodeRequired && !showUsernameModal && username !== '';
 
     return (
       <div id="page">
 
-        <Header
-          statuses={statuses}
-          showSettings={this.handleShowSettings}
-          toggleInfoModal={this.toggleInfoModal} />
-
-        <main className="encloser">
-
-          {pincodeRequired && <PincodeModal
-            showModal={pincodeRequired}
-            onSetPincode={this.onSetPincode}
-            onCloseModal={this.onClosePincodeModal} />}
-
-          {displaySettings && <UsernameModal
-            previousUsername={previousUsername}
+          <Header
```

Comment:
This is tiny, but I think this is indented one level too far?

Reply:
Yep, I'll run the linter.

Reply:
Looks good, thanks